model/artifact/artifact_file.go

```diff
@@ -113,10 +115,35 @@ func presignFile(ctx context.Context, file File) (string, error) {
 
 	// If this bucket is a devprod owned one, we sign the URL
 	// with the app's server IRSA credentials (which is used
-	// when no credentials are provided).
+	// when no credentials are provided). If it fails,
+	// we fallback to using the provided credentials.
 	if isInternalBucket(file.Bucket) {
-		file.AwsKey = ""
-		file.AwsSecret = ""
+		requestParams := pail.PreSignRequestParams{
+			Bucket:                file.Bucket,
+			FileKey:               file.FileKey,
+			SignatureExpiryWindow: evergreen.PresignMinimumValidTime,
+		}
+		presignURL, err := pail.PreSign(ctx, requestParams)
+		if err != nil {
+			return "", errors.Wrap(err, "presigning internal bucket file")
+		}
+		// Test if the presigned URL is valid by making a request to it.
+		resp, err := http.Get(presignURL)
+		if err == nil {
+			defer resp.Body.Close()
+		}
+		if err != nil || resp.StatusCode == http.StatusForbidden {
```

Comment:
Should this check for any non-200 statuses, rather than specifically forbidden?

Reply:
That sounds good, I got the forbidden from manual testing. I changed it to be any code not 200 and include it in the error if so.